### ISSUE TITLE
Sonoff Dual application_makegpioswitch = App.make_gpio_switch

### ISSUE THREAD

**Issue by dawsmac** [esphome/ESPHome#194]
Hey Guys, 

So the other day I lost all my data on my intel NUC along with my esphomeyaml yaml's, not great... so I have spend the last few days reflashing all my units, so I have a copy for when I need to do the need update/upgrades and now backed up ;)  Everything has gone fine for the 20 plus devces a mixture of D1 and Sonoff,  apart from my Sonoff Dual, When uploading the below code I get the below message, firmware upload fine but the switch's dont work. 
Any help would be great thanks :)

```
rc/main.cpp: In function 'void setup()':
src/main.cpp:22:31: warning: variable 'application_makegpioswitch' set but not used [-Wunused-but-set-variable]
Application::MakeGPIOSwitch application_makegpioswitch = App.make_gpio_switch("Downstairs radiators", 5);
^
src/main.cpp:23:31: warning: variable 'application_makegpioswitch_2' set but not used [-Wunused-but-set-variable]
Application::MakeGPIOSwitch application_makegpioswitch_2 = App.make_gpio_switch("Upstairs radiators", 12);
^
```

```yaml
esphomeyaml:
  name: rad
  platform: ESP8266
  board: esp01_1m
  board_flash_mode: dout


wifi:
  ssid: 'Automation'
  password: '*******'
  domain: '.******'

mqtt:
  broker: '*****'
  username: ''
  password: ''


switch:
  - platform: gpio
    name: "Downstairs radiators"
    pin: GPIO5
  - platform: gpio
    name: "Upstairs radiators"
    pin:
    number: GPIO12

logger:

ota:
  password: '*****'


web_server:
  port: 80 }

```

**Comment by OttoWinter** [comment]
So the message in the logs is just a warning, I've recently manually disabled that warning message and that "fix" will be live in 1.9.0

As for the issue, I unfortunately have no idea what it could be. The gpio switch platform is pretty well tested so my best bet would be that it's *probably* a hardware issue.

**Comment by dawsmac** [comment]
Hey OttoWinter, 

Cheers for clearing that up :+1:  for testing is it possible for me to test against ver 1.7 ? 

Cheers again for you help
